### PR TITLE
feat(stark-ui): date-range-picker - add support for timestamp mask

### PR DESCRIPTION
**Pull request by SuperITMan**
ISSUES CLOSED: #1231

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #549 #1231 

## What is the new behavior?
Add support for timestamp in date-range-picker

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

File: packages/stark-ui/src/modules/date-range-picker/components/date-range-picker.component.ts [hunk 1]
```diff
@@ -30,6 +34,15 @@ export class StarkDateRangePickerComponent extends AbstractStarkUiComponent impl
 	@Input()
 	public dateFilter?: StarkDatePickerFilter;
 
+	/**
+	 * Timestamp Mask Configuration to apply on the date-picker.
```
Comment (christophercr) [inline packages/stark-ui/src/modules/date-range-picker/components/date-range-picker.component.ts]: "... to apply on the start/end range date-pickers"

File: showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html [hunk 2]
```diff
@@ -10,6 +10,7 @@ <h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
 				[dateFilter]="customDateFilter"
 				[startDate]="startDate"
 				startDateLabel="STARK.DATE_RANGE_PICKER.FROM"
+				[dateMask]="true"
```
Comment (christophercr) [inline showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html]: Can you add another example for custom date masks?

File: showcase/tslint.json [hunk 3]
```diff
@@ -1,15 +0,0 @@
-{
-  "extends": [
```
Comment (christophercr) [inline showcase/tslint.json]: Why is this `tslint.json` file removed?

Comment (SuperITMan) [reply showcase/tslint.json]: Ooops! I thought I didn't take this change... My bad. I'll revert this

File: showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html (hunk 2 above)
Comment (christophercr) [reply showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html]: I would suggest to divide the examples in 2 groups:

1.- Simple date range picker (normal config with no mask... so basically what we previously had)
2.- With date masks (one date-range-picker with default mask and another date-range-picker with custom mask)

**Comment by SuperITMan** [comment]
@christophercr @cnomes I updated my PR. Could you please have a look on it ? 😊

File: showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html [hunk 4]
```diff
@@ -22,5 +22,59 @@ <h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
...
+	<example-viewer
+		[extensions]="['HTML', 'TS']"
+		filesPath="date-range-picker/date-mask"
+		exampleTitle="SHOWCASE.DEMO.DATE_RANGE_PICKER.DEFAULT_MASK"
+	>
+		<div>
+			<stark-date-range-picker
+				rangePickerId="date-mask-range-picker"
+				rangePickerName="date-mask-range-picker"
+				[isDisabled]="isDisabled"
```
Comment (christophercr) [inline showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html]: The `isDisabled` property is not included in the TS example file of this date-range-picker...

File: showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html [hunk 5]
```diff
@@ -22,5 +22,59 @@ <h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
...
+	<example-viewer
+		[extensions]="['HTML', 'TS']"
+		filesPath="date-range-picker/custom-date-mask"
+		[exampleTitle]="'SHOWCASE.DEMO.DATE_RANGE_PICKER.CUSTOM_MASK' | translate: { mask: customDateMask.format }"
+	>
+		<div>
+			<stark-date-range-picker
+				rangePickerId="date-mask-range-picker"
+				rangePickerName="date-mask-range-picker"
+				[isDisabled]="isDisabled"
```
Comment (christophercr) [inline showcase/src/app/demo-ui/pages/date-range-picker/demo-date-range-picker-page.component.html]: The `isDisabled` property is not included in the TS example file of this date-range-picker...

File: showcase/src/assets/examples/date-range-picker/custom-date-mask.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		[isDisabled]="isDisabled"
+		[dateFilter]="customDateFilter"
+		[dateMask]="customDateMask"
+		[startDate]="startDate"
+		startDateLabel="From"
+		[startMinDate]="minDate"
+		[startMaxDate]="maxDate"
+		[endDate]="endDate"
+		endDateLabel="To"
+		dateMask="true"
```
Comment (christophercr) [inline showcase/src/assets/examples/date-range-picker/custom-date-mask.html]: `dateMask` input duplicated

File: showcase/src/assets/examples/date-range-picker/custom-date-mask.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkDatePickerFilter, StarkTimestampMaskConfig } from "@nationalbankbelgium/stark-ui";
+
+@Component({
+	selector: "demo-date-range-picker",
+	templateUrl: "./demo-date-range-picker.component.html"
+})
+export class DemoDateRangePickerComponent implements OnInit {
+	public startDate: Date;
```
Comment (christophercr) [inline showcase/src/assets/examples/date-range-picker/custom-date-mask.ts]: The `isDisabled` property is missing

File: showcase/src/assets/examples/date-range-picker/date-mask.html [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+		[isDisabled]="isDisabled"
+		[dateFilter]="customDateFilter"
+		dateMask
+		[startDate]="startDate"
+		startDateLabel="From"
+		[startMinDate]="minDate"
+		[startMaxDate]="maxDate"
+		[endDate]="endDate"
+		endDateLabel="To"
+		[endMinDate]="minDate"
```
Comment (christophercr) [inline showcase/src/assets/examples/date-range-picker/date-mask.html]: The HTML example file for this date-range-picker is not in sync...

File: showcase/src/assets/examples/date-range-picker/custom-date-mask.html [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+		[endMinDate]="minDate"
```
Comment (christophercr) [inline showcase/src/assets/examples/date-range-picker/custom-date-mask.html]: The HTML example file for this date-range-picker is not in sync...

File: showcase/src/assets/examples/date-range-picker/date-mask.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkDatePickerFilter } from "@nationalbankbelgium/stark-ui";
+
+@Component({
+	selector: "demo-date-range-picker",
+	templateUrl: "./demo-date-range-picker.component.html"
+})
+export class DemoDateRangePickerComponent implements OnInit {
+	public startDate: Date;
```
Comment (christophercr) [inline showcase/src/assets/examples/date-range-picker/date-mask.ts]: The `isDisabled` property is missing

**Pull request merged**